packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -145,6 +146,10 @@ export default function parse(source: string, state: State): TemplateParseResult
     let parent: IRElement;
     const stack: IRElement[] = [];
 
+    const preserveComments =
+        templateRoot.attrs.some(({ name }) => name === PRESERVE_COMMENTS_OPTION) ||
```

Comment:
```suggestion
        templateRoot.attrs.some(({ name }) => name === PRESERVE_COMMENTS_OPTION) ??
```

Reply:
@pmdartus `.some` will always return `true` or `false` therefore the right side of `??` will never be reached right?

Reply:
My bad.